### PR TITLE
samples: Add GCS fileio samples

### PR DESCRIPTION
**Pull request by unforced**

### CODE REVIEW

**Comment by Breathtender** [comment]
This still likely needs a sample that covers some more complex things with bytes and zip and flush. @andrewsg

**Review by cojenco — COMMENTED** [review]
Thanks for adding these! Could you also update the [samples readme](https://github.com/googleapis/python-storage/blob/main/samples/README.md) to include the new samples?

**Comment by Breathtender** [comment]
> Thanks for adding these! Could you also update the [samples readme](https://github.com/googleapis/python-storage/blob/main/samples/README.md) to include the new samples?

Thanks for the reminder, added that it.

By the way, I notice that in the README it says how to run the samples but never demonstrates any args, and most of the samples will fail out if they're not passed certain arguments. Do you think that's a concern?

**Comment by cojenco** [comment]
> > Thanks for adding these! Could you also update the [samples readme](https://github.com/googleapis/python-storage/blob/main/samples/README.md) to include the new samples?
> 
> Thanks for the reminder, added that it.
> 
> By the way, I notice that in the README it says how to run the samples but never demonstrates any args, and most of the samples will fail out if they're not passed certain arguments. Do you think that's a concern?

Indeed, that was something I considered. Adding the args in the commands is definitely more clear; but I also noticed that it was more prone to error and easily out-of date with future changes when I referenced other libraries/samples. How about instead I add a note about using Cloud Shell and running the commands with the required arguments. WDYT?

**Comment by Breathtender** [comment]
> > > Thanks for adding these! Could you also update the [samples readme](https://github.com/googleapis/python-storage/blob/main/samples/README.md) to include the new samples?
> > 
> > 
> > Thanks for the reminder, added that it.
> > By the way, I notice that in the README it says how to run the samples but never demonstrates any args, and most of the samples will fail out if they're not passed certain arguments. Do you think that's a concern?
> 
> Indeed, that was something I considered. Adding the args in the commands is definitely more clear; but I also noticed that it was more prone to error and easily out-of date with future changes when I referenced other libraries/samples. How about instead I add a note about using Cloud Shell and running the commands with the required arguments. WDYT?

Ya that sounds like a nice way to do it; just some documentation somewhere so people don't get confused when the errors pop up.

File: samples/README.md [hunk 1]
```diff
@@ -486,6 +488,22 @@ View the [source code](https://github.com/googleapis/python-storage/blob/main/sa
...
 `python storage_enable_versioning.py`
 
+-----
+### FileIO Write-Read
+[![Open in Cloud Shell][shell_img]](https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/python-storage&page=editor&open_in_editor=samples/snippets/storage_fileio_write_read.py,samples/README.md)
+
+View the [source code](https://github.com/googleapis/python-storage/blob/main/samples/snippets/storage_fileio_write_read.py). To run this sample:
+
+
+`python storage_fileio_write_read.py`
```
Comment (andrewsg) [inline samples/README.md:498]: If the sample needs command-line args to work then I think we should mention that here as well.

File: samples/snippets/storage_fileio_pandas.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    with blob.open("r") as f:
+        df = pd.read_csv(f)
+        print(df.values.flatten())
+
+
+# [END storage_fileio_pandas]
+
+if __name__ == "__main__":
+    pandas_write_read(
+        bucket_name=sys.argv[1],
```
Comment (andrewsg) [inline samples/snippets/storage_fileio_pandas.py:53]: I'm not clear if best practice in python-docs-samples is to use sys.argv or a helper like argparse (I asked the same about the concurrent pubsub sample work). Can you please verify this is the standard method for Python samples these days?

File: samples/snippets/storage_fileio_pandas.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    # The ID of your new GCS object
+    # blob_name = "storage-object-name"
+
+    import pandas as pd
+
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(blob_name)
+
+    with blob.open("w") as f:
```
Comment (andrewsg) [inline samples/snippets/storage_fileio_pandas.py:40]: I think this is a good minimalist sample concept.

File: samples/snippets/storage_fileio_pandas.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import sys
+
+"""Sample that creates and consumes a GCS blob using pandas with file-like IO
+"""
+
+# [START storage_fileio_pandas]
+from google.cloud import storage
+
+
+def pandas_write_read(bucket_name, blob_name):
```
Comment (andrewsg) [inline samples/snippets/storage_fileio_pandas.py:26]: We could consider breaking this out into a write and a read sample, and then running them one after the other in `__main__`. That way users copy/pasting can easily copy/paste the writes or reads as they need them, but the command-line sample still works the same way.

**Review by andrewsg — CHANGES_REQUESTED** [review]

File: samples/README.md (hunk 1 above)
Comment (Breathtender) [reply samples/README.md:498]: Yes, done.

File: samples/snippets/storage_fileio_pandas.py (hunk 2 above)
Comment (Breathtender) [reply samples/snippets/storage_fileio_pandas.py:53]: Hmm, a little research has shown me more places tending towards argparse, so perhaps that's the standard?

File: samples/snippets/storage_fileio_pandas.py (hunk 3 above)
Comment (Breathtender) [reply samples/snippets/storage_fileio_pandas.py:40]: Do you mean by that we may not need the other hello world write-read sample at all?

File: samples/snippets/storage_fileio_pandas.py (hunk 4 above)
Comment (Breathtender) [reply samples/snippets/storage_fileio_pandas.py:26]: Good idea, on it.

File: samples/README.md (hunk 1 above)
Comment (cojenco) [reply samples/README.md]: Per our discussion, we can submit a separate PR to update all command-line args in the readme, if we decide to keep them. 

For now, could you help update the command-line args to `<BUCKET_NAME>` `<BLOB_NAME>` for consistency? Thanks!

i.e. `python storage_fileio_write_read.py <BUCKET_NAME> <BLOB_NAME>`

Comment (Breathtender) [reply samples/README.md]: Yes, done.

File: samples/snippets/storage_fileio_pandas.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (cojenco) [inline samples/snippets/storage_fileio_pandas.py:3]: Sorry missed this earlier. Copyright year should be 2021?

Comment (Breathtender) [reply samples/snippets/storage_fileio_pandas.py:3]: Woops! Good catch :)

**Review by cojenco — APPROVED** [review]
LGTM, thanks!

File: samples/snippets/storage_fileio_pandas.py (hunk 2 above)
Comment (andrewsg) [reply samples/snippets/storage_fileio_pandas.py:84]: Okay, well, given our conversation about Cloud-wide standards for samples, I'm ok with approving this as-is and we'll sort that out afterwards.

**Review by andrewsg — APPROVED** [review]
Thanks!

**Pull request merged**